Logic/Wechat.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace jean553\WechatBundle\Logic;
+
+class Wechat
+{
+    const WECHAT_AUTH2_URL =
+        "https://api.weixin.qq.com/sns/oauth2/access_token?";
+
+    /**
+     * @var string $openId Wechat openid of the user
+     */
+    private $openid;
+
+    /**
+     * Authenticate the user, check if the connection is
+     * done through the WeChat browser. Use the appid
+     * and secret passphrase to authenticate the user.
+     *
+     * @param string $appid WeChat application identifier
+     * @param string $secret WeChat application secret
+     *
+     * @return boolean authentication succeed or failure
+     */
+    public function authenticate($appid, $secret) {
+
+        $wechatCode = $this->getWeChatCode();
+
+        if(is_null($wechatCode)) {
+            return false;
+        }
+
+        // create the authentication URL with parameters
+        $url = self::WECHAT_AUTH2_URL .
+            "appid=" . $appid .
+            "&secret=" . $secret .
+            "&code=" . $wechatCode .
+            "&grant_type=authorization_code";
+
+        // call the CURL service and get the response
+        $accessToken = $this->executeCurlRequest($url);
```

Comment:
same, as above, as I thought your were authenticating your server , that's why i said access token,  because access token as a really well defined meaning in oauth2 

it's the string you give to a oauth2 server to prove that you're authorized
so here it does not apply to your "array"